sound/soc/sof/ipc.c

```diff
@@ -272,7 +296,7 @@ int sof_ipc_tx_message(struct snd_sof_ipc *ipc, u32 header,
 
 	/* schedule the message if not busy */
 	if (snd_sof_dsp_is_ready(sdev))
-		schedule_work(&ipc->tx_kwork);
+		ipc_tx_next_msg_unlock(ipc);
```

Comment:
If the DSP is busy(means transiting another ipc is in progress), we don't send ipc here, it may have chance to be sent after the one in transiting is done(at the bottom of wait_done()).

And, this logic stay no changed by my PR.